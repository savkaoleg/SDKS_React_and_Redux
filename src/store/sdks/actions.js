import { SET_SDKS } from './types'

export function setSdks(payload) {
  return {
    type: SET_SDKS,
    payload
  }
}
