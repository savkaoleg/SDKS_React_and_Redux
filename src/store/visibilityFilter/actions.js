import { HANDLE_CHANGE, HANDLE_SEARCH_CHANGE } from './types'

export function handleChange(newType, newAditional) {
  return {
    type: HANDLE_CHANGE,
    newType,
    newAditional
  }
}

export function handleSearcChange(newType, searchQuery) {
  return {
    type: HANDLE_SEARCH_CHANGE,
    newType,
    searchQuery
  }
}
