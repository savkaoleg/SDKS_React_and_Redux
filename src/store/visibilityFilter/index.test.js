import expect from 'expect'
import reducer, {initialState} from './index'
import { handleChange, handleSearcChange } from './actions'

describe('>>> visibilityFilter > reducer', ()=>{
  describe('>>> visibilityFilter > reducer > handleChange', ()=>{
    const newType = 'Tag'
    const newAditional = 'TagName'

    it('handleChange snapshot', () => {

      const next = reducer(initialState, handleChange(newType, newAditional))
      expect(next).toMatchSnapshot()
    })

    it('handleChange', () => {
      const next = reducer(initialState, handleChange(newType, newAditional))
      expect(next).toMatchObject({
        'aditional': newAditional,
        'type': newType
      })
    })
  })


  describe('>>> visibilityFilter > reducer > handleSearcChange', ()=>{
    const newType = 'Search'
    const searchQuery = 'searchQuery'

    it('handleSearcChange snapshot', () => {

      const next = reducer(initialState, handleSearcChange(newType, searchQuery))
      expect(next).toMatchSnapshot()
    })

    it('handleSearcChange snapshot', () => {
      const next = reducer(initialState, handleSearcChange(newType, searchQuery))
      expect(next).toMatchObject({
        'search': searchQuery,
        'type': newType
      })
    })
  })
})
