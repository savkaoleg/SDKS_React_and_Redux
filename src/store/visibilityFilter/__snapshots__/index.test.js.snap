// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> visibilityFilter > reducer >>> visibilityFilter > reducer > handleChange handleChange snapshot 1`] = `
Object {
  "aditional": "TagName",
  "search": "",
  "type": "Tag",
}
`;

exports[`>>> visibilityFilter > reducer >>> visibilityFilter > reducer > handleSearcChange handleSearcChange snapshot 1`] = `
Object {
  "aditional": "",
  "search": "searchQuery",
  "type": "Search",
}
`;
